platform/os_windows.go

```diff
@@ -302,32 +303,61 @@ func restartHNS(ctx context.Context) error {
 	}
 	defer service.Close()
 	// Stop the service
-	_, err = service.Control(svc.Stop)
-	if err != nil {
-		return errors.Wrap(err, "could not stop service")
+	log.Printf("Stopping HNS service")
+	_ = retry.Do(
+		tryStopServiceFn(ctx, service),
+		retry.UntilSucceeded(),
+		retry.Context(ctx),
+	)
+	// Start the service again
+	log.Printf("Starting HNS service")
+	if err := service.Start(); err != nil {
+		return errors.Wrap(err, "could not start service")
 	}
-	// Wait for the service to stop
-	ticker := time.NewTicker(500 * time.Millisecond) //nolint:gomnd // 500ms
-	defer ticker.Stop()
-	for { // hacky cancellable do-while
+	log.Printf("HNS service started")
+	return nil
+}
+
+type managedService interface {
+	Control(control svc.Cmd) (svc.Status, error)
+	Query() (svc.Status, error)
+}
+
+func tryStopServiceFn(ctx context.Context, service managedService) func() error {
+	return func() error {
 		status, err := service.Query()
 		if err != nil {
 			return errors.Wrap(err, "could not query service status")
 		}
+		// If the service is already stopped, no need to stop it again
 		if status.State == svc.Stopped {
```

Comment:
theres no stopping state right?

Reply:
there is stop pending, added

Reply:
could it potentially get stuck in the stop pending state too?

Reply:
I think it's possible, but the only thing we can safely do about that is wait